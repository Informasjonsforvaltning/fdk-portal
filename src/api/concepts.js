import axios from 'axios';
import get from 'lodash/get';
import { getConfig } from '../config';

export const conceptsUrlBase = () =>
  `${getConfig().conceptApi.host}/api/concepts`;

export const conceptsSearch = params =>
  axios
    .get(conceptsUrlBase(), {
      ...getConfig().conceptApi.config,
      params
    })
    .then(r => r.data);

export const getConcept = id =>
  axios
    .get(`${conceptsUrlBase()}/${id}`, getConfig().conceptApi.config)
    .then(r => r.data)
    .catch(() => null);

export const extractConcepts = searchResponse =>
  get(searchResponse, ['_embedded', 'concepts']);

export const extractAggregations = searchResponse =>
  get(searchResponse, 'aggregations');

export const extractTotal = searchResponse =>
  get(searchResponse, ['page', 'totalElements']);
