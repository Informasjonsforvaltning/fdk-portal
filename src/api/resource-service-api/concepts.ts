import { resourceServiceApiGet } from './host';

export const getConcepts = () => resourceServiceApiGet('/concepts');

export const paramsToSearchBody = ({ ids }: any) => ({
  ids
});
