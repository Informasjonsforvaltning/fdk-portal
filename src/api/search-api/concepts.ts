import { searchApiPost } from './host';
import { Concept } from '../../types';

export const searchConcepts = (body: any) => searchApiPost('/concepts', body);

export const extractConcepts = (searchResponse: any) =>
  searchResponse?.hits ?? [];

export const extractConceptAggregations = (searchResponse: any) =>
  searchResponse.aggregations ?? [];

export const extractConceptsTotal = (searchResponse: any) =>
  searchResponse?.page?.totalElements ?? 0;

export const extractFirstConcept = (searchResponse: any): Concept | undefined =>
  searchResponse?.hits?.[0];
