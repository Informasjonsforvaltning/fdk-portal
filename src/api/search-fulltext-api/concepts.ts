import { searchFullTextApiPost } from './host';
import { normalizeAggregations } from '../../lib/normalizeAggregations';

const mapSorting = ({ sortfield }: any) =>
  sortfield === 'modified'
    ? { field: 'harvest.firstHarvested', direction: 'desc' }
    : undefined;

const mapFilters = ({ id, identifiers, uri, orgPath, last_x_days }: any) => {
  const filters = [];
  if (id) {
    filters.push({ _id: id });
  }
  if (identifiers) {
    filters.push({
      collection: {
        field: 'identifier.keyword',
        values: identifiers
      }
    });
  }
  if (uri) {
    filters.push({
      collection: {
        field: 'uri',
        values: uri
      }
    });
  }
  if (orgPath) {
    filters.push({ orgPath });
  }
  if (last_x_days) {
    filters.push({ last_x_days });
  }
  return filters.length > 0 ? filters : undefined;
};

export const searchConcepts = (body: any) =>
  searchFullTextApiPost('/concepts', body);

export const extractConcepts = ({ hits = [] }: any) => hits;

export const extractConceptAggregations = (searchResponse: any) =>
  normalizeAggregations(searchResponse).aggregations ?? [];

export const extractConceptsTotal = (searchResponse: any) =>
  searchResponse?.page?.totalElements ?? 0;

export const paramsToSearchBody = ({ q, page, size, ...params }: any) => ({
  q,
  page: page ? Number(page) : undefined,
  size,
  sorting: mapSorting(params),
  filters: mapFilters(params)
});

export const extractFirstConcept = ({ hits = [] }: any) => hits[0] ?? {};
