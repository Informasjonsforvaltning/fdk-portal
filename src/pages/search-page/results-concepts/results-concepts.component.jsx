import React from 'react';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';
import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from 'reactstrap';
import cx from 'classnames';
import _get from 'lodash/get';
import _capitalize from 'lodash/capitalize';
import _ from 'lodash';
import { Link } from 'react-router-dom';
import { withRouter } from 'react-router';

import {
  PATHNAME_CONCEPTS,
  PATHNAME_CONCEPTS_COMPARE
} from '../../../constants/constants';
import localization from '../../../lib/localization';
import { ConceptsHitItem } from './concepts-hit-item/concepts-hit-item.component';
import { CompareTerms } from './compare-terms/compare-terms.component';
import { FilterTree } from '../filter-tree/filter-tree.component';
import { getTranslateText } from '../../../lib/translateText';
import { getSortfield, setPage, setSortfield } from '../search-location-helper';
import { parseSearchParams } from '../../../lib/location-history-helper';
import { FilterPills } from '../filter-pills/filter-pills.component';

function _renderCompareTerms({ conceptsCompare, removeConcept }) {
  const conceptIdsArray = [];
  const children = items =>
    Object.keys(items).map(el => {
      const item = items[el];
      conceptIdsArray.push(item.id);
      const { publisher } = item;
      const publisherPrefLabel =
        getTranslateText(_get(publisher, ['prefLabel'])) ||
        _capitalize(_get(publisher, 'name', ''));

      return (
        <CompareTerms
          key={item.uri}
          uri={item.id}
          prefLabel={item.prefLabel}
          creator={publisherPrefLabel}
          onDeleteTerm={removeConcept}
        />
      );
    });

  if (conceptsCompare && Object.keys(conceptsCompare).length > 0) {
    return (
      <div className="mt-5">
        <h3 className="mb-3">{localization.terms.compareTerms}</h3>
        {children(conceptsCompare)}
        <div className="d-flex justify-content-center">
          <Link
            to={`${PATHNAME_CONCEPTS}${PATHNAME_CONCEPTS_COMPARE}?compare=${conceptIdsArray}`}
          >
            {localization.compare.openCompare}
          </Link>
        </div>
      </div>
    );
  }
  return null;
}

function _renderTerms({
  conceptItems,
  conceptsCompare,
  addConcept,
  removeConcept
}) {
  if (conceptItems && Array.isArray(conceptItems)) {
    return conceptItems.map(
      item =>
        item && (
          <ConceptsHitItem
            key={item.id}
            result={item}
            concepts={conceptsCompare}
            onAddConcept={addConcept}
            onDeleteConcept={removeConcept}
          />
        )
    );
  }
  return null;
}

function _renderFilterModal({
  showFilterModal,
  closeFilterModal,
  onFilterPublisherHierarchy,
  locationSearch,
  publisherCounts,
  publishers
}) {
  return (
    <Modal isOpen={showFilterModal} toggle={closeFilterModal}>
      <ModalHeader toggle={closeFilterModal}>Filter</ModalHeader>
      <ModalBody>
        <div className="search-filters">
          <FilterTree
            title={localization.responsible}
            aggregations={publisherCounts}
            handleFiltering={onFilterPublisherHierarchy}
            activeFilter={locationSearch.orgPath}
            referenceDataItems={publishers}
          />
        </div>
      </ModalBody>
      <ModalFooter>
        <Button
          className="fdk-button-default fdk-button"
          onClick={closeFilterModal}
        >
          Close
        </Button>
      </ModalFooter>
    </Modal>
  );
}

export const ResultsConceptsPure = ({
  showFilterModal,
  closeFilterModal,
  conceptAggregations,
  conceptTotal,
  onFilterPublisherHierarchy,
  hitsPerPage,
  publisherCounts,
  publishers,
  conceptItems,
  conceptsCompare,
  addConcept,
  removeConcept,
  history,
  location
}) => {
  const locationSearch = parseSearchParams(location);

  const page = parseInt(locationSearch.page || 0, 10);
  const pageCount = Math.ceil((conceptTotal || 1) / hitsPerPage);

  const sortfield = getSortfield(location);
  const sortByScoreClass = cx('fdk-button', 'fdk-button-black-toggle', {
    selected: !sortfield
  });
  const sortByLastModifiedClass = cx('fdk-button', 'fdk-button-black-toggle', {
    selected: sortfield === 'modified'
  });

  const onSortByScoreClick = () => {
    setSortfield(history, location, undefined);
  };
  const onSortByModifiedClick = () => {
    setSortfield(history, location, 'modified');
  };

  const onPageChange = data => {
    setPage(history, location, data.selected);
    window.scrollTo(0, 0);
  };

  return (
    <main id="content">
      <section className="row mb-3">
        <div className="col-6 col-lg-4" />
        <div className="col-6 col-lg-4 offset-lg-4">
          <div className="d-flex justify-content-end">
            <Button
              className={sortByScoreClass}
              onClick={onSortByScoreClick}
              color="primary"
            >
              {localization.sort.relevance}
            </Button>
            <Button
              className={sortByLastModifiedClass}
              onClick={onSortByModifiedClick}
              color="primary"
            >
              {localization.sort.modified}
            </Button>
          </div>
        </div>
      </section>

      <section className="row">
        <aside className="search-filters col-lg-4">
          <div className="d-none d-lg-block">
            <span className="uu-invisible" aria-hidden="false">
              Filtrering tilgang
            </span>

            <FilterPills
              history={history}
              location={location}
              locationSearch={locationSearch}
              publishers={publishers}
            />

            {conceptAggregations && (
              <div>
                {_renderFilterModal({
                  showFilterModal,
                  closeFilterModal,
                  onFilterPublisherHierarchy,
                  locationSearch,
                  publisherCounts,
                  publishers
                })}
                <FilterTree
                  title={localization.responsible}
                  aggregations={publisherCounts}
                  handleFiltering={onFilterPublisherHierarchy}
                  activeFilter={locationSearch.orgPath}
                  referenceDataItems={publishers}
                />
              </div>
            )}
          </div>
          {_renderCompareTerms({ conceptsCompare, removeConcept })}
        </aside>

        <section className="col-lg-8">
          {_renderTerms({
            conceptItems,
            conceptsCompare,
            addConcept,
            removeConcept
          })}
        </section>

        <section className="col-lg-8 offset-lg-4 d-flex justify-content-center">
          <span className="uu-invisible" aria-hidden="false">
            Sidepaginering.
          </span>
          <ReactPaginate
            pageCount={pageCount}
            pageRangeDisplayed={2}
            marginPagesDisplayed={1}
            previousLabel={localization.page.prev}
            nextLabel={localization.page.next}
            breakLabel={<span>...</span>}
            breakClassName="break-me"
            containerClassName="pagination"
            onPageChange={onPageChange}
            subContainerClassName="pages pagination"
            activeClassName="active"
            forcePage={page}
            disableInitialCallback
          />
        </section>
      </section>
    </main>
  );
};

ResultsConceptsPure.defaultProps = {
  showFilterModal: false,
  closeFilterModal: _.noop,

  conceptItems: [],
  conceptTotal: 0,
  conceptAggregations: null,

  onFilterPublisherHierarchy: _.noop,

  publisherCounts: [],
  publishers: null,

  hitsPerPage: 10,

  conceptsCompare: {},
  addConcept: _.noop,
  removeConcept: _.noop,

  history: { push: _.noop },
  location: { search: '' }
};

ResultsConceptsPure.propTypes = {
  showFilterModal: PropTypes.bool,
  closeFilterModal: PropTypes.func,

  conceptItems: PropTypes.array,
  conceptTotal: PropTypes.number,
  conceptAggregations: PropTypes.object,

  onFilterPublisherHierarchy: PropTypes.func,
  publisherCounts: PropTypes.array,
  publishers: PropTypes.object,

  hitsPerPage: PropTypes.number,

  conceptsCompare: PropTypes.object,
  addConcept: PropTypes.func,
  removeConcept: PropTypes.func,

  history: PropTypes.object,
  location: PropTypes.object
};

export const ResultsConcepts = withRouter(ResultsConceptsPure);
