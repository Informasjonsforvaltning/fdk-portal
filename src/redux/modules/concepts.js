import _ from 'lodash';
import qs from 'qs';
import {
  conceptsSearch,
  extractAggregations,
  extractConcepts,
  extractTotal
} from '../../api/concepts';
import { reduxFsaThunk } from '../../lib/redux-fsa-thunk';

export const CONCEPTS_REQUEST = 'CONCEPTS_REQUEST';
export const CONCEPTS_SUCCESS = 'CONCEPTS_SUCCESS';
export const CONCEPTS_FAILURE = 'CONCEPTS_FAILURE';

const generateQueryKey = query => qs.stringify(query, { skipNulls: true });

function shouldFetch(metaState, queryKey) {
  const threshold = 60 * 1000; // seconds
  return (
    !metaState ||
    (!metaState.isFetching &&
      ((metaState.lastFetch || 0) < Date.now() - threshold ||
        metaState.queryKey !== queryKey))
  );
}

export function fetchConceptsIfNeededAction(query) {
  const queryKey = generateQueryKey(query);
  const params = { ...query, aggregations: 'orgPath' };

  return (dispatch, getState) =>
    shouldFetch(_.get(getState(), ['concepts', 'meta']), queryKey) &&
    dispatch(
      reduxFsaThunk(() => conceptsSearch(params), {
        onBeforeStart: { type: CONCEPTS_REQUEST, meta: { queryKey } },
        onSuccess: { type: CONCEPTS_SUCCESS, meta: { queryKey } },
        onError: { type: CONCEPTS_FAILURE, meta: { queryKey } }
      })
    );
}

const initialState = {};

export function conceptReducer(state = initialState, action) {
  switch (action.type) {
    case CONCEPTS_REQUEST:
      return {
        ...state,
        meta: {
          isFetching: true,
          lastFetch: null,
          queryKey: action.meta.queryKey
        }
      };
    case CONCEPTS_SUCCESS:
      return {
        ...state,
        conceptItems: extractConcepts(action.payload),
        conceptAggregations: extractAggregations(action.payload),
        conceptTotal: extractTotal(action.payload),
        meta: {
          isFetching: false,
          lastFetch: Date.now(),
          queryKey: action.meta.queryKey
        }
      };
    case CONCEPTS_FAILURE:
      return {
        ...state,
        conceptItems: null,
        conceptAggregations: null,
        conceptTotal: null,
        meta: {
          isFetching: false,
          lastFetch: null, // retry on error
          queryKey: action.meta.queryKey,
          error: action.payload
        }
      };
    default:
      return state;
  }
}
