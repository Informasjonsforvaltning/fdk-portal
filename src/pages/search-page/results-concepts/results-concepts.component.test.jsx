import React from 'react';
import { shallow } from 'enzyme';
import { ResultsConceptsPure } from './results-concepts.component';
import concepts from '../../../mock/concepts.response';
import {
  extractAggregations,
  extractConcepts,
  extractTotal
} from '../../../api/concepts';

test('should render ResultsConcepts correctly with minimum of props', () => {
  const result = shallow(<ResultsConceptsPure />);
  expect(result).toMatchSnapshot();
});

test('should render ResultsConcepts correctly with hits', () => {
  const props = {
    conceptItems: extractConcepts(concepts),
    conceptAggregations: extractAggregations(concepts),
    conceptTotal: extractTotal(concepts)
  };

  const result = shallow(<ResultsConceptsPure {...props} />);
  expect(result).toMatchSnapshot();
});
