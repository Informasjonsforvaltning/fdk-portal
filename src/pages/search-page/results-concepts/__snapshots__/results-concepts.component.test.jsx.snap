// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ResultsConcepts correctly with hits 1`] = `
<main
  id="content"
>
  <section
    className="row mb-3"
  >
    <div
      className="col-6 col-lg-4"
    />
    <div
      className="col-6 col-lg-4 offset-lg-4"
    >
      <div
        className="d-flex justify-content-end"
      >
        <Button
          className="fdk-button fdk-button-black-toggle selected"
          color="primary"
          onClick={[Function]}
          tag="button"
        >
          Relevance
        </Button>
        <Button
          className="fdk-button fdk-button-black-toggle"
          color="primary"
          onClick={[Function]}
          tag="button"
        >
          Recently published
        </Button>
      </div>
    </div>
  </section>
  <section
    className="row"
  >
    <aside
      className="search-filters col-lg-4"
    >
      <div
        className="d-none d-lg-block"
      >
        <span
          aria-hidden="false"
          className="uu-invisible"
        >
          Filtrering tilgang
        </span>
        <FilterPills
          history={
            Object {
              "push": [Function],
            }
          }
          location={
            Object {
              "search": "",
            }
          }
          locationSearch={Object {}}
          losItems={null}
          publishers={null}
          themesItems={null}
        />
      </div>
    </aside>
    <section
      className="col-lg-8"
    >
      <ConceptsHitItem
        concepts={Object {}}
        fadeInCounter={0}
        key="d092ccf9-93dd-4ef9-9a5a-d1a50f495d82"
        onAddConcept={[Function]}
        onDeleteConcept={[Function]}
        result={
          Object {
            "altLabel": Array [
              Object {
                "nb": "familiebarnehage",
              },
            ],
            "contactPoint": Object {
              "email": "begrepskatalogen@skatteetaten.no",
            },
            "definition": Object {
              "text": Object {
                "nb": "institusjon for barn som er under skolepliktig alder",
              },
            },
            "harvest": Object {
              "firstHarvested": "2019-04-29T12:18:21.525+0000",
              "lastHarvested": "2019-07-02T04:00:53.441+0000",
            },
            "hiddenLabel": Array [],
            "id": "d092ccf9-93dd-4ef9-9a5a-d1a50f495d82",
            "identifier": "http://begrepskatalogen/begrep/20b52ae3-9fe1-11e5-a9f8-e4115b280940",
            "prefLabel": Object {
              "nb": "barnehage",
            },
            "publisher": Object {
              "name": "SKATTEETATEN",
              "orgPath": "/STAT/972417807/974761076",
              "prefLabel": Object {
                "no": "SKATTEETATEN",
              },
              "uri": "http://data.brreg.no/enhetsregisteret/enhet/974761076",
            },
            "subject": Object {
              "nb": "Tredjepartsopplysninger",
            },
            "uri": "https://fellesdatakatalog.brreg.no/api/concepts/d092ccf9-93dd-4ef9-9a5a-d1a50f495d82",
          }
        }
      />
      <ConceptsHitItem
        concepts={Object {}}
        fadeInCounter={0}
        key="22ce6a6e-f629-4a75-826d-500cfc2d9e7a"
        onAddConcept={[Function]}
        onDeleteConcept={[Function]}
        result={
          Object {
            "altLabel": Array [],
            "contactPoint": Object {
              "email": "begrepskatalogen@skatteetaten.no",
            },
            "definition": Object {
              "remark": Object {
                "nb": "Tilskudd utover en månedlig maksimalbeløp skal arbeidstakeren som har barnehageplassen skatte av.",
              },
              "source": Object {
                "prefLabel": Object {
                  "nb": "Veiledningen til A-ordningen, - FSFIN §5-15-5, 2. ledd",
                },
              },
              "text": Object {
                "nb": "økonomisk tilskudd fra arbeidsgiveren til en barnehage som ikke er arbeidsgiverens bedriftsbarnehage.",
              },
            },
            "harvest": Object {
              "firstHarvested": "2019-04-29T12:17:02.338+0000",
              "lastHarvested": "2019-07-02T04:00:16.185+0000",
            },
            "hiddenLabel": Array [],
            "id": "22ce6a6e-f629-4a75-826d-500cfc2d9e7a",
            "identifier": "http://begrepskatalogen/begrep/8ea2df64-7662-11e6-a74e-7e18b36b3fd9",
            "prefLabel": Object {
              "nb": "tilskudd til barnehageplass",
            },
            "publisher": Object {
              "name": "SKATTEETATEN",
              "orgPath": "/STAT/972417807/974761076",
              "prefLabel": Object {
                "no": "SKATTEETATEN",
              },
              "uri": "http://data.brreg.no/enhetsregisteret/enhet/974761076",
            },
            "subject": Object {
              "nb": "Formues- og inntektsskatt",
            },
            "uri": "https://fellesdatakatalog.brreg.no/api/concepts/22ce6a6e-f629-4a75-826d-500cfc2d9e7a",
          }
        }
      />
      <ConceptsHitItem
        concepts={Object {}}
        fadeInCounter={0}
        key="815db2ae-b168-4d1e-9934-9a374a09a605"
        onAddConcept={[Function]}
        onDeleteConcept={[Function]}
        result={
          Object {
            "altLabel": Array [],
            "contactPoint": Object {
              "email": "begrepskatalogen@skatteetaten.no",
            },
            "definition": Object {
              "source": Object {
                "prefLabel": Object {
                  "nb": "FSFIN § 5-15-5",
                },
              },
              "text": Object {
                "nb": "skattfri fordel av plass i barnehage som arbeidsgiver gjennom eierandel eller innskudd disponerer til bruk for sine arbeidstakeres barn",
              },
            },
            "harvest": Object {
              "firstHarvested": "2019-04-29T12:16:53.227+0000",
              "lastHarvested": "2019-07-02T04:00:12.564+0000",
            },
            "hiddenLabel": Array [],
            "id": "815db2ae-b168-4d1e-9934-9a374a09a605",
            "identifier": "http://begrepskatalogen/begrep/8ea2df2d-7662-11e6-a74e-7e18b36b3fd9",
            "prefLabel": Object {
              "nb": "fordel av bedriftsbarnehageplass",
            },
            "publisher": Object {
              "name": "SKATTEETATEN",
              "orgPath": "/STAT/972417807/974761076",
              "prefLabel": Object {
                "no": "SKATTEETATEN",
              },
              "uri": "http://data.brreg.no/enhetsregisteret/enhet/974761076",
            },
            "subject": Object {
              "nb": "Formues- og inntektsskatt",
            },
            "uri": "https://fellesdatakatalog.brreg.no/api/concepts/815db2ae-b168-4d1e-9934-9a374a09a605",
          }
        }
      />
      <ConceptsHitItem
        concepts={Object {}}
        fadeInCounter={0}
        key="8f1a3814-0cde-470e-aa3e-fc30bf1835a4"
        onAddConcept={[Function]}
        onDeleteConcept={[Function]}
        result={
          Object {
            "altLabel": Array [],
            "contactPoint": Object {
              "email": "begrepskatalogen@skatteetaten.no",
            },
            "definition": Object {
              "text": Object {
                "nb": "faktisk kostnad som en gitt skattepliktig har i en gitt skattleggingsperionde for merkjøring for å bringe og hente barn ",
              },
            },
            "example": Object {
              "nb": "Bussbillett mellom hjem og barnehage mv., bompenger for omveien til og fra barnehagen",
            },
            "harvest": Object {
              "firstHarvested": "2019-04-29T12:17:16.835+0000",
              "lastHarvested": "2019-07-02T04:00:22.580+0000",
            },
            "hiddenLabel": Array [],
            "id": "8f1a3814-0cde-470e-aa3e-fc30bf1835a4",
            "identifier": "http://begrepskatalogen/begrep/46f4d715-4c6c-11e8-bb3e-005056821322",
            "prefLabel": Object {
              "nb": "faktisk kostnad for merkjøring for å bringe og hente barn",
            },
            "publisher": Object {
              "name": "SKATTEETATEN",
              "orgPath": "/STAT/972417807/974761076",
              "prefLabel": Object {
                "no": "SKATTEETATEN",
              },
              "uri": "http://data.brreg.no/enhetsregisteret/enhet/974761076",
            },
            "subject": Object {
              "nb": "Formues- og inntektsskatt",
            },
            "uri": "https://fellesdatakatalog.brreg.no/api/concepts/8f1a3814-0cde-470e-aa3e-fc30bf1835a4",
          }
        }
      />
      <ConceptsHitItem
        concepts={Object {}}
        fadeInCounter={0}
        key="30b843a4-0d19-41f7-aee3-7e79ec703d03"
        onAddConcept={[Function]}
        onDeleteConcept={[Function]}
        result={
          Object {
            "altLabel": Array [
              Object {
                "nb": "betalt barnepass",
              },
            ],
            "contactPoint": Object {
              "email": "begrepskatalogen@skatteetaten.no",
            },
            "definition": Object {
              "remark": Object {
                "nb": "Dette omfatter ikke kostnader til mat, drikke og merkjøring for å bringe og hente barn",
              },
              "source": Object {
                "prefLabel": Object {
                  "nb": "Skatte-ABC, fagemnet Foreldrefradrag",
                },
              },
              "text": Object {
                "nb": "summen for en bestemt skattleggingsperiode av en gitt skattepliktigs utgifter til tilsyn i barnets hjem, i barnepassers hjem, barnehage, i skolefritidsordning på barneskole, institusjon mv., for barn som gir rett til foreldrefradrag.",
              },
            },
            "harvest": Object {
              "firstHarvested": "2019-04-29T12:17:50.112+0000",
              "lastHarvested": "2019-07-02T04:00:37.283+0000",
            },
            "hiddenLabel": Array [],
            "id": "30b843a4-0d19-41f7-aee3-7e79ec703d03",
            "identifier": "http://begrepskatalogen/begrep/20b52ae7-9fe1-11e5-a9f8-e4115b280940",
            "prefLabel": Object {
              "nb": "oppholdskostnader ved pass og stell av barn",
            },
            "publisher": Object {
              "name": "SKATTEETATEN",
              "orgPath": "/STAT/972417807/974761076",
              "prefLabel": Object {
                "no": "SKATTEETATEN",
              },
              "uri": "http://data.brreg.no/enhetsregisteret/enhet/974761076",
            },
            "subject": Object {
              "nb": "Tredjepartsopplysninger",
            },
            "uri": "https://fellesdatakatalog.brreg.no/api/concepts/30b843a4-0d19-41f7-aee3-7e79ec703d03",
          }
        }
      />
      <ConceptsHitItem
        concepts={Object {}}
        fadeInCounter={0}
        key="d071c81d-0ae3-411a-8b47-bbdd99b87233"
        onAddConcept={[Function]}
        onDeleteConcept={[Function]}
        result={
          Object {
            "altLabel": Array [],
            "contactPoint": Object {
              "email": "begrepskatalogen@skatteetaten.no",
            },
            "definition": Object {
              "remark": Object {
                "nb": "Pass og stell av barn vil omfatte tilsyn, pleie mv. av barnet. Dette kan skje i barnets hjem, i barnepassers hjem, barnehage, i skolefritidsordning på barneskole, institusjon mv. Som pass og stell av barn regnes også fritidsordninger for barn i regi av klubber og foreninger som driver aktiviteter innenfor eget interessefelt med mindre aktivitieten er sporadisk. Dette kan f.eks. være aktiviteter innenfor idrett, teater, musikk, barnekor, kulturskole mv. Det er et vilkår at ordningen er etablert som et alternativ til skolefritidsordning på barneskole. Trening med aldersbestemt lag regnes ikke som pass og stell av barn. Pass og stell må avgrenses mot",
              },
              "source": Object {
                "prefLabel": Object {
                  "nb": "Skatte-ABC, fagemnet \\"Foreldrefradrag\\"",
                },
              },
              "text": Object {
                "nb": "ivaretakelse av barnets velbefinnede ved at en eller flere ansvarlige personer er til stede og viser omsorg for barnet",
              },
            },
            "harvest": Object {
              "firstHarvested": "2019-04-29T12:16:53.845+0000",
              "lastHarvested": "2019-07-02T04:00:12.965+0000",
            },
            "hiddenLabel": Array [],
            "id": "d071c81d-0ae3-411a-8b47-bbdd99b87233",
            "identifier": "http://begrepskatalogen/begrep/46f4d75b-4c6c-11e8-bb3e-005056821322",
            "prefLabel": Object {
              "nb": "pass og stell av barn",
            },
            "publisher": Object {
              "name": "SKATTEETATEN",
              "orgPath": "/STAT/972417807/974761076",
              "prefLabel": Object {
                "no": "SKATTEETATEN",
              },
              "uri": "http://data.brreg.no/enhetsregisteret/enhet/974761076",
            },
            "subject": Object {
              "nb": "Formues- og inntektsskatt",
            },
            "uri": "https://fellesdatakatalog.brreg.no/api/concepts/d071c81d-0ae3-411a-8b47-bbdd99b87233",
          }
        }
      />
      <ConceptsHitItem
        concepts={Object {}}
        fadeInCounter={0}
        key="5bb11bb8-806b-4c93-84f9-05da23679d1b"
        onAddConcept={[Function]}
        onDeleteConcept={[Function]}
        result={
          Object {
            "altLabel": Array [],
            "contactPoint": Object {
              "email": "begrepskatalogen@skatteetaten.no",
            },
            "definition": Object {
              "remark": Object {
                "nb": "En dagmamma skal godkjennes av kommunen i henhold til \\"Forskrift om miljørettet helsevern i barnehager og skoler m.v\\" dersom: 1) Man får godtgjøring for regelmessig tilsyn med og omsorg for barn, 2) tilbyr ukentlig oppholdstid på mer enn ti timer og 3) antall barn som er tilstede samtidig, er tre eller flere (inkludert dagmammaens egne barn).

Enten er foreldrene arbeidsgivere, eller dagmammaen er selvstendig næringsdrivende. Det er mange forenklede regler for dagmammajobb i forhold til andre arbeidsforhold. Bant annet skal det ikke betales arbeidsgiveravgift og dagmammaen trenger ikke nødvedigvis registrere seg som selvstedig næringsdrivende i Brønnøysundregistrene.",
              },
              "source": Object {
                "prefLabel": Object {
                  "nb": "snl.no",
                },
              },
              "text": Object {
                "nb": "person som passer andres barn regelmessig på dagtid, gjerne hele dager, og ofte mens foreldrene er på jobb. Passer som regel flere barn samtidig, enten hjemme hos seg selv eller hjemme hos ett av barna.",
              },
            },
            "harvest": Object {
              "firstHarvested": "2019-04-29T12:17:40.238+0000",
              "lastHarvested": "2019-07-02T04:00:32.617+0000",
            },
            "hiddenLabel": Array [],
            "id": "5bb11bb8-806b-4c93-84f9-05da23679d1b",
            "identifier": "http://begrepskatalogen/begrep/dcfb471c-e484-11e6-9cb3-005056821322",
            "prefLabel": Object {
              "nb": "dagmamma i egen bolig",
            },
            "publisher": Object {
              "name": "SKATTEETATEN",
              "orgPath": "/STAT/972417807/974761076",
              "prefLabel": Object {
                "no": "SKATTEETATEN",
              },
              "uri": "http://data.brreg.no/enhetsregisteret/enhet/974761076",
            },
            "subject": Object {
              "nb": "A-ordningen",
            },
            "uri": "https://fellesdatakatalog.brreg.no/api/concepts/5bb11bb8-806b-4c93-84f9-05da23679d1b",
          }
        }
      />
      <ConceptsHitItem
        concepts={Object {}}
        fadeInCounter={0}
        key="572c21d8-dd74-41a0-8bc1-cc9ddcf2c430"
        onAddConcept={[Function]}
        onDeleteConcept={[Function]}
        result={
          Object {
            "altLabel": Array [],
            "contactPoint": Object {
              "email": "begrepskatalogen@skatteetaten.no",
            },
            "definition": Object {
              "text": Object {
                "nb": "rent leketilbud uten de samme krav til pedagogiske personale som for barnehager",
              },
            },
            "harvest": Object {
              "firstHarvested": "2019-04-29T12:17:46.570+0000",
              "lastHarvested": "2019-07-02T04:00:35.700+0000",
            },
            "hiddenLabel": Array [],
            "id": "572c21d8-dd74-41a0-8bc1-cc9ddcf2c430",
            "identifier": "http://begrepskatalogen/begrep/20b52ae4-9fe1-11e5-a9f8-e4115b280940",
            "prefLabel": Object {
              "nb": "barnepark",
            },
            "publisher": Object {
              "name": "SKATTEETATEN",
              "orgPath": "/STAT/972417807/974761076",
              "prefLabel": Object {
                "no": "SKATTEETATEN",
              },
              "uri": "http://data.brreg.no/enhetsregisteret/enhet/974761076",
            },
            "subject": Object {
              "nb": "Tredjepartsopplysninger",
            },
            "uri": "https://fellesdatakatalog.brreg.no/api/concepts/572c21d8-dd74-41a0-8bc1-cc9ddcf2c430",
          }
        }
      />
    </section>
    <section
      className="col-lg-8 offset-lg-4 d-flex justify-content-center"
    >
      <span
        aria-hidden="false"
        className="uu-invisible"
      >
        Sidepaginering.
      </span>
      <PaginationBoxView
        activeClassName="active"
        breakClassName="break-me"
        breakLabel={
          <span>
            ...
          </span>
        }
        containerClassName="pagination"
        disableInitialCallback={true}
        disabledClassName="disabled"
        forcePage={0}
        marginPagesDisplayed={1}
        nextClassName="next"
        nextLabel="Neste side"
        onPageChange={[Function]}
        pageCount={1}
        pageRangeDisplayed={2}
        previousClassName="previous"
        previousLabel="Forrige side"
        subContainerClassName="pages pagination"
      />
    </section>
  </section>
</main>
`;

exports[`should render ResultsConcepts correctly with minimum of props 1`] = `
<main
  id="content"
>
  <section
    className="row mb-3"
  >
    <div
      className="col-6 col-lg-4"
    />
    <div
      className="col-6 col-lg-4 offset-lg-4"
    >
      <div
        className="d-flex justify-content-end"
      >
        <Button
          className="fdk-button fdk-button-black-toggle selected"
          color="primary"
          onClick={[Function]}
          tag="button"
        >
          Relevance
        </Button>
        <Button
          className="fdk-button fdk-button-black-toggle"
          color="primary"
          onClick={[Function]}
          tag="button"
        >
          Recently published
        </Button>
      </div>
    </div>
  </section>
  <section
    className="row"
  >
    <aside
      className="search-filters col-lg-4"
    >
      <div
        className="d-none d-lg-block"
      >
        <span
          aria-hidden="false"
          className="uu-invisible"
        >
          Filtrering tilgang
        </span>
        <FilterPills
          history={
            Object {
              "push": [Function],
            }
          }
          location={
            Object {
              "search": "",
            }
          }
          locationSearch={Object {}}
          losItems={null}
          publishers={null}
          themesItems={null}
        />
      </div>
    </aside>
    <section
      className="col-lg-8"
    />
    <section
      className="col-lg-8 offset-lg-4 d-flex justify-content-center"
    >
      <span
        aria-hidden="false"
        className="uu-invisible"
      >
        Sidepaginering.
      </span>
      <PaginationBoxView
        activeClassName="active"
        breakClassName="break-me"
        breakLabel={
          <span>
            ...
          </span>
        }
        containerClassName="pagination"
        disableInitialCallback={true}
        disabledClassName="disabled"
        forcePage={0}
        marginPagesDisplayed={1}
        nextClassName="next"
        nextLabel="Neste side"
        onPageChange={[Function]}
        pageCount={1}
        pageRangeDisplayed={2}
        previousClassName="previous"
        previousLabel="Forrige side"
        subContainerClassName="pages pagination"
      />
    </section>
  </section>
</main>
`;
